OneCommunityGlobal/HighestGoodNetworkApp
# yihan Create way to manually add in lost hours for past years of volunteers

# Description
<img width="751" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/048457ef-fee3-497f-820e-241e764fddcb">

## Related PRS (if any):
This frontend PR is related to the [#611](https://github.com/OneCommunityGlobal/HGNRest/pull/611) backend PR.

## Main changes explained:
- Create AddLostTime.jsx for introducing Add Lost Time modal.
- Create LostTimeHistory.jsx and EditHistoryModal.jsx for introducing tracking method of additions of lost time.
- Update Reports.jsx for including AddLostTime and LostTimeHistory component on reports page.
- Update TotalProjectReport.jsx, TotalPersonReport.jsx and TotalTeamReport.jsx for adding lost time to reports.

## How to test:
1. check into current branch on both FE and BE
2. do `npm run start:local` to run this PR locally
3. log as Owner user
a. go to dashboard→ Reports→ Reports, verify if the Owner user can see the "Add Lost Time" and "show project/person/team lost time" at the bottom of the page. 
b. Click on "Add Lost Time" button and add time for a project/person/team, then go to "Show Total Project/Person/Team Report", see if the time is added to the report correctly.
c. Then, click on "show project/person/team lost time", verify if the user can see, edit and delete the addition.
4. log as Admin user
a. go to dashboard→ Reports→ Reports, verify if the user can only see "show project/person/team lost time" at the bottom of the page.
b.  Click on "show project/person/team lost time", verify if the user can see, edit and delete the addition.
5. log as Manager/Mentor/Core team user
a. go to dashboard→ Reports→ Reports, verify if the user can only see "show project/person/team lost time" at the bottom of the page.
b.  Click on "show project/person/team lost time", verify if the user can see, but cannot edit or delete the addition.

## Screenshots or videos of changes:

Owner can add, edit and delete lost time:
<img width="1024" alt="Screenshot 2023-11-20 at 11 40 24" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/6816f128-48a8-47f7-86b8-cc67d200fb0e">

Administrator can edit and delete lost time:
<img width="1052" alt="Screenshot 2023-11-20 at 11 39 47" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/5d57ca3f-e3f2-4b4e-a967-8e55197364fa">

Manager/Mentor/Core team can only see the lost time:
<img width="1017" alt="Screenshot 2023-11-20 at 11 41 02" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/db82e087-a68d-4722-a37d-1acf669bbf90">

Example of testing Project: 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/e4910cd8-3879-4d20-851a-563e2c45a969

Steps of testing Person and Team are same.

## Note:
An error of "currentTarget" might occur when showing Total Project Report, which also happens on dev branch and is out of scope of this PR:
<img width="1512" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/e5494613-a3b3-4a42-9475-2a6263a42b4a">
